src/Core/src/AbstractApi.php

```diff
@@ -144,12 +144,21 @@ protected function getSignerFactories(): array
     }
 
     /**
-     * @psalm-suppress InvalidNullableReturnType
+     * This method is a BC layer for client that does not require core:^1.2.
```

Comment:
should we deprecate it ?

Reply:
New clients will implement this. Like `EventBridgeClient::getEndpointMetadata()`. The method should be declared abstract, but it has been added here to allow old versions of clients to still work as expected. 

So, no, we should not deprecate it. Is that right @jderusse?

Reply:
Yes. I think that once every clients will requires `core:^1.2` should can make it abstract.
Maybe we can trigger a deprecation telling `this method should not be called, update your client`

note: With this PR, we don't need the 3 others abstract methods (`getSignatureScopeName`, `getSignatureVersion` and `getServiceCode`)

Reply:
> Yes. I think that once every clients will requires core:^1.2 should can make it abstract.

No, we can't. We always need to consider that other people may have created their own clients. 

> Maybe we can trigger a deprecation telling this method should not be called, update your client

Yes! That sounds like a good idea.

Reply:
Well, then we should indeed deprecate the implementation in core, so that they get a warning telling them that their client must be updated to implement this method.

Making it abstract will be possible only in 2.0 as that's a BC break (but the deprecation warning will warn them before).